mat/vector.go

```diff
@@ -72,11 +72,21 @@ func (v *Vector) Dims() (r, c int) {
 	return v.n, 1
 }
 
+// Caps returns the number of rows and columns in the backing matrix. Columns is always 1.
+func (v *Vector) Caps() (r, c int) {
+	return v.Cap(), 1
+}
+
 // Len returns the length of the vector.
 func (v *Vector) Len() int {
 	return v.n
 }
 
+// Cap returns the capacity of the vector.
+func (v *Vector) Cap() int {
+	return (cap(v.mat.Data)-1)/v.mat.Inc + 1
```

Comment:
Good catch. And we can't know what a future inc will be, so returning 0 is appropriate.

The documentation is tricky. I'm really leaning towards the inclusion of IsZero methods as per gonum/matrix#396 even if only because it allows us to then define a usable zero state formally.